internal/controller/nonadminbackupstoragelocation_controller_test.go

```diff
@@ -188,19 +189,22 @@ var _ = ginkgo.Describe("Test full reconcile loop of NonAdminBackupStorageLocati
 		nonAdminBslName = fmt.Sprintf("non-admin-bsl-object-%v", counter)
 		nonAdminBslNamespace = fmt.Sprintf("test-non-admin-bsl-reconcile-full-%v", counter)
 		oadpNamespace = nonAdminBslNamespace + "-oadp"
+		ctx, cancel = context.WithCancel(context.Background())
 	})
 
 	ginkgo.AfterEach(func() {
 		gomega.Expect(deleteTestNamespaces(ctx, nonAdminBslNamespace, oadpNamespace)).To(gomega.Succeed())
 
 		cancel()
 		// wait cancel
-		time.Sleep(1 * time.Second)
+		<-ctx.Done()
+		gomega.Expect(ctx.Err()).To(gomega.Equal(context.Canceled))
```

Comment:
From initial debugging, I had made this change thought something took longer.. but I think its ok to keep. <-ctx.Done() seemed more accurate than hardcoded waits.